registry/api/protocol/eureka/instance_handler.go

```diff
@@ -232,7 +200,7 @@ func (routes *Routes) deregisterInstance(w rest.ResponseWriter, r *rest.Request)
 			"error":     "catalog is nil",
 		}).Errorf("Failed to deregister instance %s", uid)
 
-		i18n.Error(r, w, http.StatusBadRequest, i18n.ErrorNamespaceNotFound)
+		i18n.Error(r, w, http.StatusInternalServerError, i18n.ErrorNamespaceNotFound)
```

Comment:
If namespace not found, I would assume a bad request (unless you want to reserve that to malformed requests...).
Don't think it is a server error, maybe want to return 404?
This may be applicable in other error returns elsewhere in this change set.